tests/helpers/component-object/builder.ts

```diff
@@ -0,0 +1,340 @@
+	 * {@link DOM.elNameSelectorGenerator}
+	 *
+	 * @example
+	 * ```typescript
+	 * this.elSelector('element') // .${componentName}__element
+	 * ```
+	 */
+	get elSelector(): (elName: string) => string {
+		return DOM.elNameSelectorGenerator(this.componentName);
+	}
+
+	/**
+	 * Public access to the reference of the component's `JSHandle`
+	 * @throws {@link ReferenceError} if trying to access a component that has not been built or picked
+	 */
+	get component(): JSHandle<COMPONENT> {
+		if (!this.componentStore) {
+			throw new ReferenceError('Bad access to the component without "build" or "pick" call');
+		}
+
+		return this.componentStore;
+	}
+
+	/**
+	 * Returns `true` if the component is built or picked
+	 */
+	get isBuilded(): boolean {
+		return Boolean(this.componentStore);
+	}
+
+	/**
+	 * @param page - the page on which the component is located
+	 * @param componentName - the name of the component to be rendered
+	 */
+	constructor(page: Page, componentName: string) {
+		this.pwPage = page;
+		this.componentName = componentName;
+		this.id = `${this.componentName}_${Math.random().toString()}`;
+		this.props = {'data-component-object-id': this.id};
+		this.node = page.locator(`[data-component-object-id="${this.id}"]`);
+		this.componentClassImportPath = path.join(
+			path.relative(`${process.cwd()}/src`, resolve.blockSync(this.componentName)!),
+			`/${this.componentName}.ts`
+		);
+	}
+
+	/**
+	 * Returns the base class of the component
+	 */
+	async getComponentClass(): Promise<JSHandle<new () => COMPONENT>> {
+		const {componentClassImportPath} = this;
+
+		if (componentClassImportPath == null) {
+			throw new Error('Missing component path');
+		}
+
+		const
+			classModule = await Utils.import<{default: new () => COMPONENT}>(this.pwPage, componentClassImportPath),
+			classInstance = await classModule.evaluateHandle((ctx) => ctx.default);
+
+		return classInstance;
+	}
+
+	/**
+	 * Renders the component with the previously set props and children
+	 * using the `withProps` and `withChildren` methods.
+	 *
+	 * @param [options]
+	 */
+	async build(options?: BuildOptions): Promise<JSHandle<COMPONENT>> {
+		await this.insertComponentStyles();
+
+		const
+			name = this.componentName,
+			fullComponentName = `${name}${options?.functional && !name.endsWith('-functional') ? '-functional' : ''}`;
+
+		const normalizeChildren = () => {
+			const result: Dictionary<RenderContent | RenderContentFn | string> = {};
+
+			const normalizeVNode = (vNode) => ({...vNode, tag: vNode.type});
+
+			Object.entries(this.children).forEach(([slotName, child]) => {
+				if (child == null || Object.isString(child) || Object.isFunction(child)) {
+					result[slotName] = Object.cast(child);
+					return
+				}
+
+				if (Array.isArray(child)) {
+					result[slotName] = Object.cast(child.map(normalizeVNode));
+				}
+
+				result[slotName] = normalizeVNode(child);
+			});
+
+			return result;
+		}
```

Comment:
eslint перестал поддерживать блокнот?)))

```suggestion
		};
```